101-mongodb-sharded-on-centos/scripts/config_primary.sh

```diff
@@ -0,0 +1,173 @@
+#!/bin/bash
+
+mongoAdminUser=$1
+mongoAdminPasswd=$2
+location=$3
+dnsNamePrefix=$4
+
+install_mongo3() {
+    #install
+	wget https://repo.mongodb.org/yum/redhat/7/mongodb-org/3.6/x86_64/RPMS/mongodb-org-server-3.6.17-1.el7.x86_64.rpm
+	wget https://repo.mongodb.org/yum/redhat/7/mongodb-org/3.6/x86_64/RPMS/mongodb-org-shell-3.6.17-1.el7.x86_64.rpm
+	rpm -i mongodb-org-server-3.6.17-1.el7.x86_64.rpm
+	rpm -i mongodb-org-shell-3.6.17-1.el7.x86_64.rpm
+	PATH=$PATH:/usr/bin; export PATH
+
+	#disable selinux
+	sed -i 's/^SELINUX=.*/SELINUX=disabled/' /etc/sysconfig/selinux
+	sed -i 's/^SELINUX=.*/SELINUX=disabled/' /etc/selinux/config
+	setenforce 0
+
+	#kernel settings
+	if [[ -f /sys/kernel/mm/transparent_hugepage/enabled ]];then
+		echo never > /sys/kernel/mm/transparent_hugepage/enabled
+	fi
+	if [[ -f /sys/kernel/mm/transparent_hugepage/defrag ]];then
+		echo never > /sys/kernel/mm/transparent_hugepage/defrag
+	fi
+
+	#configure
+	sed -i 's/\(bindIp\)/#\1/' /etc/mongod.conf
+	
+}
+
+yum install wget -y
+fqdn="${dnsNamePrefix}.${location}.cloudapp.azure.com"
```

Comment:
this cannot be hardcoded - must be passed into the script using reference()

Reply:
I have hardcoded the suffix for now since I am only supporting public cloud. Will take care of this once I add support for other clouds

Reply:
this is not optional